Foundation/NSLock.swift

```diff
@@ -258,3 +302,59 @@ open class NSCondition: NSObject, NSLocking {
     
     open var name: String?
 }
+
+private func timeSpecFrom(date: Date) -> timespec? {
+    let interval = date.timeIntervalSince1970
+    guard interval >= 0 else {
+        return nil
+    }
+
+    var ts = timespec()
+    ts.tv_sec = Int(interval.rounded(.down))
+    ts.tv_nsec = Int((interval - Double(ts.tv_sec)) * 1_000_000_000.0)
+    return ts
+}
+
+#if os(OSX) || os(iOS)
+
+private func allocateTimedLockData() -> UnsafeMutablePointer<_timed_lock_data>? {
+    let tld = UnsafeMutablePointer<_timed_lock_data>.allocate(capacity: 1)
+    if pthread_mutex_init(_CFGetTimedLockMutex(tld), nil) == 0 {
+        if pthread_cond_init(_CFGetTimedLockCond(tld), nil) == 0 {
+            return tld
+        } else {
+            pthread_mutex_destroy(_CFGetTimedLockMutex(tld))
```

Comment:
need to check the error code

Reply:
not needed; the failure cases are EINVAL for a corrupted lock, or EBUSY which only happens in the cases of PTHREAD_MUTEX_ERRORCHECK

Reply:
but we want to know this, right? What if we have memory corruption or something. We should just `preconditon()` or at the very least `assert()` that the return value is 0.